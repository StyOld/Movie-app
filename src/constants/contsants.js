export const UPDATE_AUTH = 'UPDATE_AUTH';
export const LOGOUT = 'LOGOUT';
export const TOGGLE_LOGINFORM = 'TOGGLE_LOGINFORM';
export const HIDE_LOGINFORM = 'HIDE_LOGINFORM';
export const GET_MOVIE_DETAILS = 'GET_MOVIE_DETAILS';
export const UPDATE_MOVIE_DETAILS = 'UPDATE_MOVIE_DETAILS';
export const FETCHING_MOVIES = 'FETCHING_MOVIES';
export const UPDATE_MOVIES = 'UPDATE_MOVIES';
export const ERROR_GET_MOVIES = 'ERROR_GET_MOVIES';
export const GET_BY_TYPE_MOVIES = 'GET_BY_TYPE_MOVIES';
export const GET_GENRES_LIST = 'GET_GENRES_LIST';
export const ERROR_GET_GENRES = 'ERROR_GET_GENRES';
export const TOGGLE_DROP_DOWN = 'TOGGLE_DROP_DOWN';
export const ERROR_DELETE_SESSION = 'ERROR_DELETE_SESSION';
export const CHANGE_PAGE = 'CHANGE_PAGE';
export const CLEAR_FILTERS = 'CLEAR_FILTERS';
export const CHANGE_FILTERS = 'CHANGE_FILTERS';
export const GET_BY_TYPE_MOVIE_DETAILS = 'GET_BY_TYPE_MOVIE_DETAILS';
export const ERROR_GET_BY_TYPE_MOVIE_DETAILS = 'ERROR_GET_BY_TYPE_MOVIE_DETAILS';