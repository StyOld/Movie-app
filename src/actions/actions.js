import * as constants from '../constants/contsants'
import CallApi from '../api/api';
import Cookies from 'universal-cookie';
const cookies = new Cookies();

// export const actionCreatorUpdateAuth = dispatch(payload => {
//     return {
//         type: 'UPDATE_AUTH',
//         payload
//     }
// });

// export const actionCreatorUpdateAuth = payload => dispatch => {
//         return dispatch({
//         type: 'UPDATE_AUTH',
//         payload
//     })
// };

export const actionCreatorUpdateAuth = (payload) => {
    cookies.set('session_id', payload.session_id, {
        path: '/',
        maxAge: 2592000
    });
    return {
        type: constants.UPDATE_AUTH,
        payload
    }
};

export const actionCreatorOnLogOut = () => {
    return {
        type: constants.LOGOUT
    }
};

export const actionCreatorToggleLoginForm = () => {
    return {
        type: constants.TOGGLE_LOGINFORM
    }
};

export const actionCreatorHideLoginForm = () => {
    return {
        type: constants.HIDE_LOGINFORM
    }
};

export const actionCreatorGetMovieDetails = (payload) => {
    return {
        type: constants.GET_MOVIE_DETAILS,
        payload
        // payload: {
        //     data: payload.data
        // }
    }
};

export const actionCreatorUpdateMovie = () => {
  return {
      type: constants.UPDATE_MOVIE
  }
};

export const actionCreatorGetMovies = ({filters, page}) => {
    const {sort_by, primary_release_year, genres} = filters;

    const queryStringParams = {
        sort_by: sort_by,
        page: page,
        primary_release_year: primary_release_year
        // with_genres: genres.join(',') словил баг на стороне сервака
    };

    if (genres.length>0) queryStringParams.with_genres = genres.join(',');

    return dispatch => {
        dispatch({
            type: constants.FETCHING_MOVIES
        });

        CallApi.get('/discover/movie', {
            params: queryStringParams
        })
            .then(data => {
                dispatch({
                    type: constants.UPDATE_MOVIES,
                    payload: data
                    // payload: data.results
                });
                dispatch({
                    type: constants.GET_TOTAL_PAGE,
                    payload: data.total_pages
                })
            })
            .catch(error => {
                dispatch({
                    type: constants.ERROR_GET_MOVIES,
                    payload: error
                });
            });
    }
};

export const actionCreatorGetByTypeMovies = ({userId, params, type}) => {
    return dispatch => {
        CallApi.get(`/account/${userId}/${type}/movies`, {
            params: params
        })
            .then(data => {
                dispatch({
                    type: constants.GET_BY_TYPE_MOVIES,
                    payload: {
                        type,
                        data: data.results
                    }
                })
            })
            .catch(error => {
                dispatch({
                    type: constants.ERROR_GET_MOVIES,
                    payload: error
                })
            })
    }
    // CallApi.get(`/account/{account_id}/${type}/movies`,
};

export const actionCreatorGetGenresList = () => {
    return dispatch => {
        CallApi.get('/genre/movie/list')
            .then(data => {
                dispatch({
                    type: constants.GET_GENRES_LIST,
                    payload: {
                        data: data.genres
                    }
                })
            })
            .catch(error => {
                dispatch({
                    type: constants.ERROR_GET_GENRES,
                    payload: error
                })
            })
    }
};

export const actionCreatorToggleDropDown = () => {
    return {
        type: constants.TOGGLE_DROP_DOWN
    }
};

export const actionCreatorDeleteSession = (params) => {
    return dispatch => {
        CallApi.delete('/authentication/session', {
            params: params
        })
            .then(() => {
                dispatch({
                    type: constants.LOGOUT
                })
            })
            .catch(error => {
                dispatch({
                    type: constants.ERROR_DELETE_SESSION,
                    payload: error
                })
            })
    }
};

export const actionCreatorChangePage = (payload) => {
    return {
        type: constants.CHANGE_PAGE,
        payload
    }
};

export const actionCreatorGetTotalPage = (payload) => {
    return {
        type: constants.GET_TOTAL_PAGE,
        payload
    }
};

export const actionCreatorClearFilters = () => {
    return {
        type: constants.CLEAR_FILTERS
    }
};

export const actionCreatorChangeFilters = (event) => {
    return {
        type: constants.CHANGE_FILTERS,
        payload: {
            name: event.target.name,
            value: event.target.value
        // payload
        }
    }
};

export const actionCreatorChangeGenres = payload => {
    return dispatch => {
        dispatch({
            type: payload.target.checked ? constants.CHECKED_GENRE : constants.UNCHECKED_GENRE,
            payload: payload.target.value
        });
    };
// (payload.target.checked) ? (
//     dispatch({
//         type: 'CHECKED_GENRE',
//         payload
//     })
//     ) : (
//     dispatch({
//         type: 'UNCHECKED_GENRE',
//         payload
//     })
//     )
};

export const actionCreatorGetAccount = (params) => {
    return dispatch => {
        if (params.session_id) {
            CallApi.get('/account', {
                params: params
            })
                .then((user) => {
                    cookies.set('session_id', params.session_id, {
                        path: '/',
                        maxAge: 2592000
                    });
                    dispatch({
                        type: constants.UPDATE_AUTH,
                        payload: {
                            user,
                            session_id: params.session_id
                        }
                    })
                })
        }
    }
};