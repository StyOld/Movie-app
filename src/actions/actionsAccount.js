import * as constants from '../constants/contsants'
import CallApi from '../api/api';

export const actionCreatorGetByTypeMovies = ({userId, params, type}) => {
    return dispatch => {
        CallApi.get(`/account/${userId}/${type}/movies`, {
            params: params
        })
            .then(data => {
                dispatch({
                    type: constants.GET_BY_TYPE_MOVIES,
                    payload: {
                        type,
                        data: data.results
                    }
                })
            })
            .catch(error => {
                dispatch({
                    type: constants.ERROR_GET_MOVIES,
                    payload: error
                })
            })
    }
};