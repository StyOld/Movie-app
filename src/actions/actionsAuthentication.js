import * as constants from '../constants/contsants'
import CallApi from '../api/api';
import Cookies from 'universal-cookie';
const cookies = new Cookies();

export const actionCreatorUpdateAuth = ({user, session_id, isAuth}) => {
    cookies.set('session_id', session_id, {
        path: '/',
        maxAge: 2592000
    });
    return {
        type: constants.UPDATE_AUTH,
        payload: {
            user,
            session_id,
            isAuth
        }
    }
};

export const actionCreatorOnLogOut = () => {
    return {
        type: constants.LOGOUT
    }
};

export const actionCreatorToggleLoginForm = () => {
    return {
        type: constants.TOGGLE_LOGINFORM
    }
};

export const actionCreatorHideLoginForm = () => {
    return {
        type: constants.HIDE_LOGINFORM
    }
};

export const actionCreatorToggleDropDown = () => {
    return {
        type: constants.TOGGLE_DROP_DOWN
    }
};

export const actionCreatorDeleteSession = (params) => {
    return dispatch => {
        CallApi.delete('/authentication/session', {
            params: params
        })
            .then(() => {
                dispatch({
                    type: constants.LOGOUT
                })
            })
            .catch(error => {
                dispatch({
                    type: constants.ERROR_DELETE_SESSION,
                    payload: error
                })
            })
    }
};

export const actionCreatorGetAccount = (params) => {
    return dispatch => {
        if (params.session_id) {
            CallApi.get('/account', {
                params: params
            })
                .then((user) => {
                    cookies.set('session_id', params.session_id, {
                        path: '/',
                        maxAge: 2592000
                    });
                    dispatch({
                        type: constants.UPDATE_AUTH,
                        payload: {
                            user,
                            session_id: params.session_id
                        }
                    })
                })
        }
    }
};