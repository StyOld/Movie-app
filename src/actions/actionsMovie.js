import * as constants from '../constants/contsants'
import CallApi from "../api/api";

export const actionCreatorGetMovieDetails = (payload) => {
    return {
        type: constants.GET_MOVIE_DETAILS,
        payload
    }
};

export const actionCreatorUpdateMovieDetails = () => {
    return {
        type: constants.UPDATE_MOVIE_DETAILS
    }
};

export const actionCreatorGetByTypeMovieDetails = ({movieId, type}) => {
    return dispatch => {
        CallApi.get(`/movie/${movieId}/${type}`)
            .then(data => {
                dispatch({
                    type: constants.GET_BY_TYPE_MOVIE_DETAILS,
                    payload: {
                        type,
                        data
                    }
                })
            })
            .catch(error => {
                dispatch({
                    type: constants.ERROR_GET_BY_TYPE_MOVIE_DETAILS,
                    payload: error
                })
            })
    }
};