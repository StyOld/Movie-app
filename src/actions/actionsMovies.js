import * as constants from '../constants/contsants'
import CallApi from '../api/api';

export const actionCreatorGetMovies = ({filters, page}) => {
    const {sort_by, primary_release_year, genres} = filters;

    const queryStringParams = {
        sort_by: sort_by,
        page: page,
        primary_release_year: primary_release_year
    };

    if (genres.length>0) queryStringParams.with_genres = genres.join(',');

    return dispatch => {
        dispatch({
            type: constants.FETCHING_MOVIES
        });

        CallApi.get('/discover/movie', {
            params: queryStringParams
        })
            .then(data => {
                dispatch({
                    type: constants.UPDATE_MOVIES,
                    payload: data
                });
            })
            .catch(error => {
                dispatch({
                    type: constants.ERROR_GET_MOVIES,
                    payload: error
                });
            });
    }
};

export const actionCreatorGetGenresList = () => {
    return dispatch => {
        CallApi.get('/genre/movie/list')
            .then(data => {
                dispatch({
                    type: constants.GET_GENRES_LIST,
                    payload: {
                        data: data.genres
                    }
                })
            })
            .catch(error => {
                dispatch({
                    type: constants.ERROR_GET_GENRES,
                    payload: error
                })
            })
    }
};

export const actionCreatorChangePage = (payload) => {
    return {
        type: constants.CHANGE_PAGE,
        payload
    }
};

export const actionCreatorClearFilters = () => {
    return {
        type: constants.CLEAR_FILTERS
    }
};

export const actionCreatorChangeFilters = (event) => {
    return {
        type: constants.CHANGE_FILTERS,
        payload: {
            name: event.target.name,
            value: event.target.value
        }
    }
};