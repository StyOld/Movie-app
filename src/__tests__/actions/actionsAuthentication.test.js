import {actionCreatorUpdateAuth, actionCreatorOnLogOut} from '../../actions/actionsAuthentication';
import * as constants from '../../constants/contsants';

describe(`actionsAuthentication`, () => {
    test(`actionCreatorUpdateAuth`, () => {
        const payload = {
            user: '1',
            session_id: '1',
            isAuth: true
        };

        const result = actionCreatorUpdateAuth(payload);
        console.log(result);

        // expect(result.type).toBe(constants.UPDATE_AUTH);
        // expect(result.payload.user).toBe(payload.user);
        // expect(result.payload.session_id).toBe(payload.session_id);
        // expect(result.payload.isAuth).toBe(payload.isAuth);

        expect(result).toMatchSnapshot();

        // ({
        //     type: constants.UPDATE_AUTH,
        //     payload: payload
        // });
    })
})