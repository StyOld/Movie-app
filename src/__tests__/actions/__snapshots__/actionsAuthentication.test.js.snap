// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actionsAuthentication actionCreatorUpdateAuth 1`] = `
Object {
  "payload": Object {
    "isAuth": true,
    "session_id": "1",
    "user": "1",
  },
  "type": "UPDATE_AUTH",
}
`;
